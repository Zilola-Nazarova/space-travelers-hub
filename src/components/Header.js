import React from 'react';
import { NavLink } from 'react-router-dom';
import Planet from '../assets/planet.png';
import '../styles/Header.css';

function Header() {
  return (
    <>
      <header>
        <nav>
          <div className="logo">
            <img src={Planet} alt="planet" />
            <h1>Space Travelers&apos; Hub</h1>
          </div>
          <div className="links">
            <NavLink to="/">Rockets</NavLink>
            <NavLink to="/missions">Missions</NavLink>
            <span className="line2" />
            <NavLink to="/profile">My Profile</NavLink>
          </div>
        </nav>
        <div className="line" />
      </header>
    </>
  );
}

export default Header;
