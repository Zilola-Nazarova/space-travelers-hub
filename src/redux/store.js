import { configureStore } from '@reduxjs/toolkit';
import { logger } from 'redux-logger';
import missionsReducer from './missions/missionsSlice';
import rocketsReducer from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
    rockets: rocketsReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
});

export default store;
