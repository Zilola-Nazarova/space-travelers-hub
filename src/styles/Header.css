* {
  margin: 0;
  font-family: Arial, sans-serif;
}

nav {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 10px 60px;
}

.logo {
  display: flex;
  align-items: center;
}

.logo img {
  width: 50px;
  height: 50px;
  margin-right: 10px;
}

.links {
  display: flex;
  gap: 20px;
  list-style: none;
}

.links a {
  text-decoration: none;
}

.links a.active {
  text-decoration: underline;
}

.line2 {
  border: 1px solid #585b5e;
}

.line {
  width: 92%;
  border-bottom: 1px solid #e8e8e8;
  margin: 0 50px;
}
