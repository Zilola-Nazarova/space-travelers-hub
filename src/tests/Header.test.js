import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Header from '../components/Header';

test('Test Header', () => {
  const tree = render(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
