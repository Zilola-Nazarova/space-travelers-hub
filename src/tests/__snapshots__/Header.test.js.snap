// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <nav>
          <div
            class="logo"
          >
            <img
              alt="planet"
              src="planet.png"
            />
            <h1>
              Space Travelers' Hub
            </h1>
          </div>
          <div
            class="links"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Rockets
            </a>
            <a
              class=""
              href="/missions"
            >
              Missions
            </a>
            <span
              class="line2"
            />
            <a
              class=""
              href="/profile"
            >
              My Profile
            </a>
          </div>
        </nav>
        <div
          class="line"
        />
      </header>
    </div>
  </body>,
  "container": <div>
    <header>
      <nav>
        <div
          class="logo"
        >
          <img
            alt="planet"
            src="planet.png"
          />
          <h1>
            Space Travelers' Hub
          </h1>
        </div>
        <div
          class="links"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
          <a
            class=""
            href="/missions"
          >
            Missions
          </a>
          <span
            class="line2"
          />
          <a
            class=""
            href="/profile"
          >
            My Profile
          </a>
        </div>
      </nav>
      <div
        class="line"
      />
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
